### PR TITLE
py-apache-beam: Submission with dependencies

### PR DESCRIPTION
**Pull request by essandess**
py-apache-beam: Submission with dependencies

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.x
Xcode 8.x

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-apache-beam/Portfile [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                    SDKs for constructing pipelines and Runners for\
+                    executing them on distributed processing backends,\
+                    including Apache Apex, Apache Flink, Apache Spark,\
+                    Google Cloud Dataflow and Hazelcast Jet.
+
+homepage            https://beam.apache.org
+master_sites        pypi:a/apache-beam
+use_zip             yes
+extract.mkdir       no
+distname            apache-beam-${version}
```
Comment (mf2k) [inline python/py-apache-beam/Portfile]: Use this instead:

`distname            ${python.rootname}-${version}`

File: python/py-apache-beam/Portfile [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+checksums           rmd160  dc243db6c9fffab8cc82f1d62822fb3eae5b983b \
+                    sha256  1843d0cfc9f945e90d21e3d94b59e82613d0d5cbc44ae7befa65872f114e6b94 \
+                    size    1621014
+
+python.versions     27 35 36 37 38
```
Comment (mf2k) [inline python/py-apache-beam/Portfile]: Let's only go with the latest python versions for new ports. So remove py27, 35, and 36.

File: python/py-avro/Portfile [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+description         Avro is a serialization and RPC framework.
+long_description    ${description}
+
+homepage            https://avro.apache.org/
+
+checksums           rmd160  61c45d1cfa9914306401cb88d880ac63602d43cc \
+                    sha256  f456539183df0d6912daedfbf0f94d7cf69b4fba7200490e8a06d1e8f23af43d \
+                    size    1845163
+
+python.versions     27 35 36 37 38
```
Comment (mf2k) [inline python/py-avro/Portfile]: Please remove py27, 35 and 36.

File: python/py-grpcio-tools/Portfile [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+homepage            https://grpc.io
+master_sites        pypi:g/grpcio-tools
+distname            grpcio-tools-${version}
+
+checksums           rmd160  54c0aed4605621845469b5c14ed10f5617f9cecb \
+                    sha256  5580b86cf49936c9c74f0def44d3582a7a1bb720eba8a14805c3a61efa790c70 \
+                    size    1990162
+
+python.versions     27 35 36 37 38
```
Comment (mf2k) [inline python/py-grpcio-tools/Portfile]: Please remove py27, 35 and 36.

File: python/py-grpcio-tools/Portfile [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+platforms           darwin
+license             Apache-2
+maintainers         nomaintainer
+
+description         Protobuf code generator for gRPC
+long_description    ${description}
+
+homepage            https://grpc.io
+master_sites        pypi:g/grpcio-tools
+distname            grpcio-tools-${version}
```
Comment (mf2k) [inline python/py-grpcio-tools/Portfile]: Use `python.rootname` here.

File: python/py-nose_xunitmp/Portfile [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+homepage            https://github.com/Ignas/nose_xunitmp
+
+master_sites        pypi:n/${python.rootname}/
+distname            ${python.rootname}-${version}
+
+checksums           rmd160  92c0a9a7f9e6fa9a9c8ddcff212c49063e59439d \
+                    sha256  c2df72f4760751d30153d473822aabf1a7c3d462f664aa7f7fdb8dc626dc7c40 \
+                    size    2885
+
+python.versions     27 35 36 37 38
```
Comment (mf2k) [inline python/py-nose_xunitmp/Portfile]: Please remove py27, 35 and 36.

File: python/py-tenacity/Portfile [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+homepage            https://github.com/jd/tenacity
+master_sites        pypi:t/tenacity
+distname            tenacity-${version}
+
+checksums           rmd160  8e5f3e8813d5f39cd34cef1c46edc7490b74a037 \
+                    sha256  72f397c2bb1887e048726603f3f629ea16f88cb3e61e4ed3c57e98582b8e3571 \
+                    size    33589
+
+python.versions     27 35 36 37
```
Comment (mf2k) [inline python/py-tenacity/Portfile]: Please remove py27, 35 and 36. 

Why no 38 version? Without it, the other ports will not support a py38 sub-port.

File: python/py-apache-beam/Portfile (hunk 1 above)
Comment (essandess) [reply python/py-apache-beam/Portfile]: Thanks done.

File: python/py-apache-beam/Portfile (hunk 2 above)
Comment (essandess) [reply python/py-apache-beam/Portfile]: Thanks done.

File: python/py-avro/Portfile (hunk 3 above)
Comment (essandess) [reply python/py-avro/Portfile]: Thanks done.

File: python/py-grpcio-tools/Portfile (hunk 4 above)
Comment (essandess) [reply python/py-grpcio-tools/Portfile]: Thanks done.

File: python/py-grpcio-tools/Portfile (hunk 5 above)
Comment (essandess) [reply python/py-grpcio-tools/Portfile]: Thanks done.

File: python/py-tenacity/Portfile (hunk 7 above)
Comment (essandess) [reply python/py-tenacity/Portfile]: Thanks done, with `38`.

File: python/py-apache-beam/Portfile [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+                    port:py${python.version}-pytest \
+                    port:py${python.version}-pytest-xdist \
+                    port:py${python.version}-requests-mock \
+                    port:py${python.version}-tenacity \
+                    port:py${python.version}-yaml
+
+    if { ${python.version} == 27 } {
+        depends_run-append \
+                    port:py${python.version}-typing
+    }
```
Comment (mf2k) [inline python/py-apache-beam/Portfile]: This block is no longer necessary and should be removed.

File: python/py-arrow/Portfile [hunk 9]
```diff
@@ -18,7 +18,7 @@ checksums           rmd160  4a19d9c8792ac47646cb7603502b3bb9a53e1fa7 \
                     sha256  543cc4a4f4e10c20412dd1d5d0c1d10d351b9874b404303b3dd82a88d1a8138e \
                     size    68029
 
-python.versions     27 36 37
+python.versions     27 35 36 37 38
```
Comment (mf2k) [inline python/py-arrow/Portfile]: Please don't add old python versions (py35) unless they are needed for a specific reason.

File: python/py-crcmod/Portfile [hunk 10]
```diff
@@ -20,7 +20,7 @@ checksums           rmd160  503b415394d7c833b22e5a999454c0d67598cd54 \
                     sha256  dc7051a0db5f2bd48665a990d3ec1cc305a466a77358ca4492826f41f283601e \
                     size    89670
 
-python.versions     35 36 37 38
+python.versions     27 35 36 37 38
```
Comment (mf2k) [inline python/py-crcmod/Portfile]: Please don't add `py27` here.

File: python/py-apache-beam/Portfile (hunk 8 above)
Comment (essandess) [reply python/py-apache-beam/Portfile]: Done thanks.

File: python/py-arrow/Portfile (hunk 9 above)
Comment (essandess) [reply python/py-arrow/Portfile]: Done thanks.

File: python/py-crcmod/Portfile (hunk 10 above)
Comment (essandess) [reply python/py-crcmod/Portfile]: Done thanks. I also removed old Python versions.

**Comment by mf2k** [comment]
@essandess Can you fix resolve the conflicts please?

**Comment by essandess** [comment]
> Can you resolve the conflicts please?

I think I got it.

**Pull request merged**